internal/webhooks/patch_validation.go

```diff
@@ -198,6 +199,36 @@ func validateSelectors(selector clusterv1.PatchSelector, class *clusterv1.Cluste
 	if selector.MatchResources.MachineDeploymentClass != nil && len(selector.MatchResources.MachineDeploymentClass.Names) > 0 {
 		for i, name := range selector.MatchResources.MachineDeploymentClass.Names {
 			match := false
+			if strings.Contains(name, "*") {
+				// selector can at most have a single * rune
+				if strings.Count(name, "*") > 1 {
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"selector can at most contain a single \"*\" rune"))
+					break
+				}
+
+				// the * rune can appear only at the beginning, or ending of the selector.
+				if strings.Contains(name, "*") && !(strings.HasPrefix(name, "*") || strings.HasSuffix(name, "*")) {
+					// templateMDClass can only have "*" rune at the start or end of the string
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"\"*\" rune can only appear at the beginning, or ending of the selector"))
+					break
+				}
+				// a valid selector with/out "*" should comply with Kubernetes naming standards.
+				if validation.IsQualifiedName(strings.ReplaceAll(name, "*", "a")) != nil {
+					allErrs = append(allErrs, field.Invalid(
+						path.Child("matchResources", "machineDeploymentClass", "names").Index(i),
+						name,
+						"selector does not comply with the Kubernetes naming standards"))
+					break
+				}
+				match = true
+				break
```

Comment:
```suggestion
```
I think we don't actually check if the selector matches here.

I think the validation here is fine but we should drop match = true and break

Reply:
I think we have to extend l.233 with the new matching logic (essentially use the same logic as in json_patch_generator instead of just a simple `==`)